### PR TITLE
Home page: use our own illustrations

### PR DESCRIPTION
**Pull request by AdrienClairembault**
## Checklist before requesting a review

- [X] I have performed a self-review of my code.
- [X] I have added tests that prove my fix is effective or that my feature works.

## Description

- Remove undraw.
- Add our own illustrations into `/pics/illustrations`.
- Add a new twig function to easily use the illustrations: `{{ render_illustration("my_icon.svg", 100) }}`.
- The illustrations are adapted to the current user theme, this is done by injecting css variables into the SVG before it is rendered.
- Use the illustrations for the home page tiles.

## Screenshots (if appropriate):

Note: icon are still being worked on and will keep being improved in further iterations.

image

### CODE REVIEW

File: src/Glpi/UI/IllustrationManager.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+        return $svg_content;
+    }
+
+    private function getSvgContent(string $filename): string
+    {
+        $svg_content = file_get_contents(GLPI_ROOT . "/pics/illustration/$filename");
+        if (!$svg_content) {
+            trigger_error("Unknown illustration: $filename", E_USER_WARNING);
+
```
Comment (cedric-anne) [inline src/Glpi/UI/IllustrationManager.php:58]: `file_get_contents` will anyway trigger a warning if the file content cannot be fetched.

```suggestion
```

File: src/Glpi/UI/IllustrationManager.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    public const DEFAULT_ICON = "request-service.svg";
+
+    /**
+     * @param int $size Height and width (px)
+     */
+    public function render(string $filename, int $size = 100): string
+    {
+        $svg_content = $this->getSvgContent($filename);
+        $svg_content = $this->replaceColorsByVariables($svg_content);
+        $svg_content = $this->setSize($svg_content, $size);
```
Comment (cedric-anne) [inline src/Glpi/UI/IllustrationManager.php:48]: This method is not really a getter.
```suggestion
        $svg_content = $this->adjustSize($svg_content, $size);
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by orthagh — APPROVED** [review]

**Comment by AdrienClairembault** [comment]
Converted to draft as I've just received a new iteration of some icons.
I'll update it quickly.

**Review by cedric-anne — APPROVED** [review]
Seems OK, but you have conflicts to resolve.

**Pull request merged**